### DISCUSSION TITLE
Design Patterns JavaScript

### DISCUSSION THREAD

**Discussion by fabiootc** [frontendbr/forum#147] [Perguntas e Respostas]
Alguém poderia me recomendar algum Design Patterns de JavaScript e/ou bom material de leitura? 

Atualmente trabalho com e-commerce, e gostaria de deixar meu código mais organizado e principalmente leve!

Ps: @Mendrone já viu minhas bagunças!! kkkk

**Comment by mandado** [comment]
Opa! @fabiootc já viu esse ? hehe 

https://addyosmani.com/resources/essentialjsdesignpatterns/book/

**Comment by fabiootc** [comment]
@mandado Ainda não, dei uma olhada bem por cima. Vlw a indicação!

**Comment by vitor-mendrone-deel** [comment]
Já vi mesmo, o rapaz é rei da lógica mas organizar e padronizar a parada passa longe! HAHAHAH 

O livro do Osmani é imbatível mas também gosto muito desse artigo: 

http://alistapart.com/article/the-design-of-code-organizing-javascript

Uso bastante o Module Pattern em códigos mais simples pra manter a parada organizada.

**Comment by abnersajr** [comment]
A minha dúvida quase sempre é em um site simples, projeto de agência padrão, alguns plugins ali, umas functions pra algun ajuste de height ali e tal como organizar isso. Geralmente eu faço varias functions <name> (){}

E chamo elas dentro do ready, load e etc.
Talvez tenha algum pattern melhor, mas eu não consigo entender e ver qual seria o melhor a usar. 

Dicas?

**Comment by diessica** [comment]
@abnersajr Dá uma olhada nesses links:
- https://css-tricks.com/how-do-you-structure-javascript-the-module-pattern-edition/
- https://css-tricks.com/how-do-you-structure-javascript-the-module-pattern-edition/#comment-253473 (um comentário específico no post acima)

E no artigo que o @Mendrone postou.

É bacana pra começar a visualizar Design Patterns no mundo real de "projetos de agência padrão". :) 

Eu vi um post muito bom no CodePen sobre isso também, mas eu perdi. :(

**Comment by diessica** [comment]
Tem também este no CSS Tricks:
https://css-tricks.com/can-you-over-organize-javascript/
